node/src/accountant/mod.rs

```diff
@@ -1435,6 +1580,317 @@ mod tests {
+        let system = System::new("test");
+        let peer_actors = peer_actors_builder()
+            .blockchain_bridge(blockchain_bridge)
+            .build();
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
+        let ui_message = NodeFromUiMessage {
+            client_id: 1234,
+            body: UiScanRequest {
+                scan_type: ScanType::Receivables,
+            }
+            .tmb(4321),
+        };
+
+        subject_addr.try_send(ui_message).unwrap();
+
+        System::current().stop();
+        system.run();
+        let blockchain_bridge_recording = blockchain_bridge_recording_arc.lock().unwrap();
+        assert_eq!(
+            blockchain_bridge_recording.get_record::<RetrieveTransactions>(0),
+            &RetrieveTransactions {
+                recipient: make_wallet("earning_wallet"),
+                response_skeleton_opt: Some(ResponseSkeleton {
+                    client_id: 1234,
+                    context_id: 4321
+                }),
+            }
+        );
+    }
+
+    #[test]
+    fn received_payments_with_response_skeleton_sends_response_to_ui_gateway() {
+        let config = bc_from_ac_plus_earning_wallet(
+            AccountantConfig {
+                scan_intervals: ScanIntervals {
+                    payable_scan_interval: Duration::from_millis(10_000),
+                    receivable_scan_interval: Duration::from_millis(10_000),
+                    pending_payable_scan_interval: Duration::from_secs(100),
+                },
+                when_pending_too_long_sec: DEFAULT_PENDING_TOO_LONG_SEC,
+                suppress_initial_scans: true,
+                payment_thresholds: *DEFAULT_PAYMENT_THRESHOLDS,
+            },
+            make_wallet("earning_wallet"),
+        );
+        let subject = AccountantBuilder::default()
+            .bootstrapper_config(config)
+            .build();
+        let (ui_gateway, _, ui_gateway_recording_arc) = make_recorder();
+        let subject_addr = subject.start();
+        let system = System::new("test");
+        let peer_actors = peer_actors_builder().ui_gateway(ui_gateway).build();
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
+        let received_payments = ReceivedPayments {
+            payments: vec![],
+            response_skeleton_opt: Some(ResponseSkeleton {
+                client_id: 1234,
+                context_id: 4321,
+            }),
+        };
+
+        subject_addr.try_send(received_payments).unwrap();
+
+        System::current().stop();
+        system.run();
+        let ui_gateway_recording = ui_gateway_recording_arc.lock().unwrap();
+        assert_eq!(
+            ui_gateway_recording.get_record::<NodeToUiMessage>(0),
+            &NodeToUiMessage {
+                target: MessageTarget::ClientId(1234),
+                body: UiScanResponse {}.tmb(4321),
+            }
+        );
+    }
+
+    #[test]
+    fn scan_payables_request() {
```

Comment:
I think you would be able to combine the body of the test with other test of this type as `scan_receivables_request`. It can have shared major part of it and it can be parametrized.